### PR TITLE
Rewrite the python warehouse schedule task body in SQL

### PR DESCRIPTION
**Pull request by joshelser**
Not nearly complete. Only does the selection of schedules to consider, none of the building up of `alter warehouse` commands.

On top of #381

### CODE REVIEW

**Comment by joshelser** [comment]
thinking aloud:
* I want to be able to pass in a `timestamp` from outside, else writing tests to cover these procedures is nary impossible.
* Passing in a `TIMESTAMP_LTZ` (which is what is defined in the task table) appears to result in mangled dates where the original wall clock time is converted to UTC wall clock time, but the timezone is left as is (e.g. if default timezone is -7, the wall clock gets 7hours added to it but timezone still claims -7).
* The warehouse schedules are defined without timezones, but need to be shifted with the opscenter defined timezone.
* When the task runs, it needs to pick the appropriate schedule(s) based off of the last time the task ran and "now".
* Timestamp_ltz's appear to be reasonable _within_ the stored procedure. So maybe this is just an issue with injecting timestamps into the procedure for testing?

**Comment by joshelser** [comment]
In a nutshell:

* Every time the warehouse schedules are changed, we generate `ALTER WAREHOUSE` statements for each schedule in the system and store them in a table, referenced back to the `wh_schedules` table with the same `id_val` column.
* SQL proc takes the current time and the last time the task ran, and sees if there are any schedules which should run now.
* For every schedule found, we `execute immediate ..` the `alter warehouse` statement and record an outcome into the task_warehouse_schedule table.
* A very simple unit test (that is a bit contrived as we're lacking the admin procedures for CRUD of warehouse schedules)

File: bootstrap/011_warehouse_schedules.sql [hunk 1]
```diff
@@ -1,18 +1,96 @@
...
-$$
-from crud import WarehouseSchedulesTask
-def run(session):
-    task = WarehouseSchedulesTask(session)
-    return task.run()
-$$;
+DECLARE
+    task_outcome variant default (select object_construct());
+BEGIN
+    -- The task calls this procedure with NULL and lets the procedure figure out the details.
```
Comment (rymurr) [inline bootstrap/011_warehouse_schedules.sql:13]: i wonder if this might be neater using a session parameter?

**Review by rymurr — APPROVED** [review]
one minor comment hat can be addressed later

File: bootstrap/011_warehouse_schedules.sql (hunk 1 above)
Comment (joshelser) [reply bootstrap/011_warehouse_schedules.sql:13]: Rather than passing in `NULL`, you mean? I didn't think we could do that in an owner's rights procedure.

Comment (rymurr) [reply bootstrap/011_warehouse_schedules.sql:13]: that null is hella ugly so we have to think of something to do.

**Pull request merged**